File: src/main/java/net/onelitefeather/attollo/Attollo.java
package net.onelitefeather.attollo;

import net.onelitefeather.attollo.listener.ElevatorListener;
import net.onelitefeather.attollo.service.ElevatorService;
import net.onelitefeather.attollo.service.ElevatorServiceImpl;
import org.bukkit.plugin.java.JavaPlugin;

public final class Attollo extends JavaPlugin {

    private final ElevatorService elevatorService = new ElevatorServiceImpl(this);

    @Override
    public void onLoad() {
        saveDefaultConfig();
    }

    @Override
    public void onEnable() {
        this.elevatorService.loadBlockFromConfig();
        getServer().getPluginManager().registerEvents(new ElevatorListener(this.elevatorService), this);
    }
}

File: src/main/kotlin/dev/themeinerlp/attollo/Attollo.kt
package dev.themeinerlp.attollo

import dev.themeinerlp.attollo.listener.AttolloListener
import dev.themeinerlp.attollo.listener.UpdateCheckerListener
import dev.themeinerlp.attollo.service.UpdateService
import org.bukkit.Material
import org.bukkit.plugin.java.JavaPlugin

open class Attollo : JavaPlugin() {

    lateinit var elevatorBlock: Material
    lateinit var updateService: UpdateService

    override fun onLoad() {
        saveDefaultConfig()
    }

    override fun onEnable() {
        val rawValue = config.getString("elevatorBlock") ?: "DAYLIGHT_DETECTOR"
        val material = Material.matchMaterial(rawValue)

        if (material == null) {
            logger.warning("Invalid elevatorBlock material in config.yml: '$rawValue'. Defaulting to DAYLIGHT_DETECTOR.")
        }

        elevatorBlock = material ?: Material.DAYLIGHT_DETECTOR
        logger.info("Using elevatorBlock: $elevatorBlock")
        server.pluginManager.registerEvents(AttolloListener(this), this)
        server.pluginManager.registerEvents(UpdateCheckerListener(this), this)
        updateChecker()
    }

    override fun onDisable() {
        updateService.shutdown()
    }

    private fun updateChecker() {
        updateService = UpdateService(this)
        updateService.run()
        updateService.notifyConsole(componentLogger)
    }

}